File: update_faq.py
import json
import re

with open('data.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Build Sidebar
sidebar_html = """<aside class="w-full md:w-64 flex-shrink-0">
<div class="sticky top-24 bg-surface-container-lowest rounded-xl shadow-[0_4px_20px_rgb(0,0,0,0.04)] p-md border border-outline-variant">
<h3 class="font-headline-md text-headline-md text-on-surface mb-sm">Danh mục</h3>
<nav class="flex flex-col gap-2">"""

for i, cat in enumerate(data['categories']):
    active_class = 'bg-primary text-white hover:bg-primary/80 active-tab' if i == 0 else 'text-on-surface-variant hover:bg-surface-container-low'
    icon_hidden = '' if i == 0 else ' hidden'
    sidebar_html += f"""
<a class="faq-tab-btn font-label-md text-label-md px-4 py-3 rounded-lg {active_class} flex items-center justify-between transition-colors" data-target="{cat['id']}" href="javascript:void(0)">
    {cat['title']}
    <span class="material-symbols-outlined text-sm{icon_hidden}" data-icon="chevron_right">chevron_right</span>
</a>"""

sidebar_html += """
</nav>
</div>
</aside>"""

# Build Accordion Content
content_html = """<div class="flex-grow">"""

for i, cat in enumerate(data['categories']):
    hidden_class = ' block' if i == 0 else ' hidden'
    content_html += f"""
<!-- Category: {cat['title']} -->
<div class="faq-tab-content{hidden_class} scroll-mt-24" id="{cat['id']}">
<h2 class="font-headline-lg text-headline-lg-mobile md:font-headline-lg md:text-headline-lg text-on-surface mb-md flex items-center gap-xs">
<span class="material-symbols-outlined text-primary" data-icon="{cat['icon']}">{cat['icon']}</span>
                            {cat['title']}
                        </h2>
<div class="space-y-4">"""

    for qa in cat['qas']:
        content_html += f"""
<!-- Question Item -->
<div class="bg-surface-container-lowest rounded-xl shadow-[0_4px_20px_rgb(0,0,0,0.04)] border border-outline-variant overflow-hidden">
<button class="accordion-header w-full px-md py-4 flex justify-between items-center text-left hover:bg-surface-container-low transition-colors" onclick="toggleAccordion(this)">
<span class="font-label-md text-label-md text-on-surface">{qa['q']}</span>
<span class="material-symbols-outlined text-outline-variant accordion-icon" data-icon="expand_more">expand_more</span>
</button>
<div class="accordion-content">
<div class="accordion-inner px-md pb-4 pt-2 font-body-md text-body-md text-on-surface-variant border-t border-outline-variant">
{qa['a']}
</div>
</div>
</div>"""

    content_html += """
</div>
</div>"""

content_html += """
</div>"""

full_section_html = f"""<section class="pt-0 pb-xl max-w-5xl mx-auto px-6 md:px-12">
<div class="flex flex-col md:flex-row gap-lg">
{sidebar_html}
{content_html}
</div>
</section>"""


with open('faq.html', 'r', encoding='utf-8') as f:
    html = f.read()

pattern = re.compile(r'<section class="pt-0 pb-xl max-w-5xl mx-auto px-6 md:px-12">.*?</section>', re.DOTALL)
new_html = pattern.sub(full_section_html, html, count=1)

with open('faq.html', 'w', encoding='utf-8') as f:
    f.write(new_html)

print("FAQ updated successfully.")

File: update_faq.js
const fs = require('fs');
let content = fs.readFileSync('faq.html', 'utf-8');

// 1. Search Bar Updates
const oldSearchBar = `<div class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none">
                            <span class="material-symbols-outlined text-outline" data-icon="search">search</span>
                        </div>
                        <input id="faq-search-input" class="w-full pl-12 pr-4 py-4 rounded-full border-outline-variant focus:border-primary focus:ring-2 focus:ring-primary/20 font-body-md text-body-md bg-transparent" placeholder="Bạn cần tìm gì? (VD: Lãi suất vay, Thủ tục...)" type="text">
                        <button id="faq-search-btn" class="absolute inset-y-1 right-1 bg-primary text-on-primary font-label-md text-label-md px-6 rounded-full hover:bg-surface-tint transition-colors">
                            Tìm kiếm
                        </button>`;

const newSearchBar = `<div class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none">
                            <span class="material-symbols-outlined text-outline" data-icon="search">search</span>
                        </div>
                        <input id="faq-search-input" class="w-full pl-12 pr-14 py-4 rounded-full border-outline-variant focus:border-primary focus:ring-2 focus:ring-primary/20 font-body-md text-body-md bg-transparent" placeholder="Bạn cần tìm gì?" type="text">
                        <button id="faq-search-btn" class="absolute inset-y-1 right-1 bg-primary text-on-primary w-12 flex items-center justify-center rounded-full hover:bg-surface-tint transition-colors">
                            <span class="material-symbols-outlined">search</span>
                        </button>`;

content = content.replace(oldSearchBar, newSearchBar);

// 2. Categories Dropdown for Mobile
const oldNav = `<nav class="flex flex-col gap-2">`;
const newNav = `<select id="faq-category-select" class="lg:hidden w-full px-4 py-3 bg-primary text-white rounded-lg border-none font-label-md text-label-md outline-none mb-4 cursor-pointer appearance-none bg-[url('data:image/svg+xml;utf8,<svg fill=%22white%22 height=%2224%22 viewBox=%220 0 24 24%22 width=%2224%22 xmlns=%22http://www.w3.org/2000/svg%22><path d=%22M7 10l5 5 5-5z%22/></svg>')] bg-no-repeat bg-[position:right_10px_center]">
                            <option value="doi-tuong">Đối tượng thụ hưởng</option>
                            <option value="dieu-kien-ho-so">Điều kiện, Hồ sơ & Quy trình</option>
                            <option value="vay-von">Vay vốn & Tài chính</option>
                            <option value="quyen-so-huu">Quyền sở hữu, Chuyển nhượng & Pháp lý</option>
                        </select>
                        <nav class="hidden lg:flex flex-col gap-2">`;

content = content.replace(oldNav, newNav);

// Also change the width of the aside to only be w-64 on lg
content = content.replace('<aside class="w-full md:w-64 flex-shrink-0">', '<aside class="w-full lg:w-64 flex-shrink-0">');

fs.writeFileSync('faq.html', content, 'utf-8');
console.log("Updated faq.html for search bar and category dropdown.");
